staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -48,3 +48,52 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta
+
+	// ConnectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService
+}
+
+// ConnectionService provides the configuration for a single network proxy client.
+type ConnectionService struct {
+	// Name is the name of the connectivity service.
+	// Currently supported values are "Master", "Etcd" and "Cluster"
```

Comment:
Can we lowercase them?

Instead of Master/Cluster, how about control-plane-nodes/nodes? Or something along those lines?

Reply:
I tolower() on consumption so case is mostly irrelevant.

Reply:
These names were specified in the KEP. (https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190226-network-proxy.md#kubernetes-api-server-outbound-requests) Not sure how we feel about modifying what we specified there without reflecting it in the KEP.